sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1615,6 +1634,22 @@ def _provision(self,
                 # TPU.
                 tpu_create_script=config_dict.get('tpu-create-script'),
                 tpu_delete_script=config_dict.get('tpu-delete-script'))
+
+            # Get actual zone info and save it into handle
+            get_zone_cmd = handle.launched_resources.cloud.get_zone_shell_cmd()
+            if get_zone_cmd is not None:
+                # Leave the zone field to None for multi-node cases
+                # if zone is not specified because head and worker nodes
+                # can be launched in different zones.
+                if (task.num_nodes > 1 and
+                        handle.launched_resources.zone is None):
```

Comment:
Done. for the second comment, right but the query can be a sanity check for `ray up`. I think it's still useful?

Reply:
The sanity check may increase the overhead of the `sky launch` on an existing cluster, but checking it may be useful for catching problems, up to you.

If the zone mismatch, that probably means an error occurs, right? Should we fail directly instead? One possible situation the error may occur is when the user has a cluster stopped in `us-west-1a`, but when she tries to `sky start` the cluster, a resources unavailable error occurs on that zone, and `ray up` starts a new cluster on `us-west-1b`. I think in that case, we should consider terminate the newly launched instance (not sure if the old stopped instance will be terminated by ray up), but preserve the cluster entry in our database so that the user can retry and launch the instance in the previous zone again.

Reply:
Oh just realized I missed this comment.. for 2, I thought `sky start` would just fail due to resources unavailable error? it shouldn't failover to other zone right or I misunderstand it?

Reply:
Had offline discussion with @Michaelvll , the above case can happen also with the current master and should be considered a bug to fix. Will file an issue.

Reply:
Filed an issue here. https://github.com/skypilot-org/skypilot/issues/1054 Will continue tracking this in the future.